adoc/chapters/programming_interface.adoc

```diff
@@ -1422,1390 +1422,1969 @@ info::context::atomic_fence_scope_capabilities
+Deprecated by SYCL 2020.
 
-a@
-[source]
-----
-info::device::partition_max_sub_devices
-----
+{note}Use [api]#device::has# with one of the [code]#aspect::usm_*# aspects
+instead.
+{endnote}
 
-    @ [.code]#uint32_t#
-   a@ Returns the maximum number of sub-devices that can be created when this SYCL [code]#device# is partitioned. The value returned cannot exceed the value returned by [code]#info::device::max_compute_units#.
+_Remarks:_ Template parameter to [api]#device::get_info#.
 
-a@
-[source]
-----
-info::device::partition_properties
-----
+_Returns:_ The value [coee]#true# if the device and the host have a unified
+memory subsystem and returns [code]#false# otherwise.
 
-    @ [.code]#std::vector<info::partition_property>#
-   a@ Returns the partition properties supported by this SYCL [code]#device;# a
-      vector of [code]#info::partition_property#.  An element is returned in
-      this vector only if the device can be partitioned into at least two sub
-      devices along that partition property.
+'''
 
-a@
-[source]
+.[apidef]#info::device::atomic_memory_order_capabilities#
+[source,role=synopsis,id=api:info-device-atomic-memory-order-capabilities]
 ----
-info::device::partition_affinity_domains
+namespace sycl::info::device {
+struct atomic_memory_order_capabilities {
+  using return_type = std::vector<memory_order>;
+};
+} // namespace sycl::info::device
 ----
 
-    @ [.code]#std::vector<info::partition_affinity_domain>#
-   a@ Returns a [code]#std::vector# of the partition affinity domains
-      supported by this SYCL [code]#device# when partitioning with
-      [code]#info::partition_property::partition_by_affinity_domain#.
-      An element is returned in this vector only if the device can be
-      partitioned into at least two sub devices along that affinity domain.
+_Remarks:_ Template parameter to [api]#device::get_info#.
 
-a@
-[source]
-----
-info::device::partition_type_property
-----
+_Returns:_ The set of memory orders supported by atomic operations on this
+device.
+When a device returns a "stronger" memory order in this set, it must also return
+all "weaker" memory orders.
+(See <<sec:memory-ordering>> for a definition of "stronger" and "weaker" memory
+orders.)
+The memory orders [code]#memory_order::acquire#, [code]#memory_order::release#,
+and [code]#memory_order::acq_rel# are all the same strength.
+If a device returns one of these, it must return them all.
 
-    @ [.code]#info::partition_property#
-   a@ Returns the partition property of this SYCL [code]#device#. If this SYCL [code]#device# is not a sub device then the return value must be [code]#info::partition_property::no_partition#, otherwise it must be one of the following values:
---
-  * [code]#info::partition_property::partition_equally#
-  * [code]#info::partition_property::partition_by_counts#
-  * [code]#info::partition_property::partition_by_affinity_domain#
---
+At a minimum, each device must support [code]#memory_order::relaxed#.
```

Comment:
Is this should be a `__remaks_` or ` __postcondition__` ? ( without thinking to much, all the "must" may be in a `condition`)

Reply:
Oh, I see. Thanks for the explanation!